python/pyarrow/array.pxi

```diff
@@ -2812,6 +2805,22 @@ cdef dict _array_classes = {
 }
 
 
+cdef inline shared_ptr[CBuffer] c_mask_from_obj(object mask, MemoryPool pool) except *:
```

Comment:
Maybe use "inverted" in the name of the function, so that when reading the code in `from_arrays`, you still have some notion about the fact that the mask gets inverted